src/bluesky/callbacks/best_effort.py

```diff
@@ -598,25 +633,60 @@ def check_visibility(self):
             else:
                 for artist in self.__arts:
                     artist.set_visible(True)
+                    label = artist.get_label()
+                    if label.startswith("_"):
+                        artist.set_label(label[1:])
```

Comment:
@mrakitin I think the parametrization needs to happen either for the number of artists instances or labels for each artist, most likely the former. Or would we keep all the lines on the graph but have the legend limited?

we could trim both the lines and the legend, I guess?